code/data_structures/tree/path_sum/path_sum.hpp

```diff
@@ -0,0 +1,48 @@
+/*
+ Part of Cosmos by OpenGenus Foundation
+ */
+
+#ifndef path_sum_hpp
+#define path_sum_hpp
+
+#include <memory>
+#include <vector>
+#include <functional>
+#include "utility/tree_node.cpp"
+
+template<typename _Ty,
+         typename _Compare = std::equal_to<_Ty>,
+         class _TreeNode = TreeNode<_Ty>>
+class PathSum
+{
+private:
+    template<typename _T>
+    using Node = _TreeNode;
+    template<typename _T>
+    using PNode = std::shared_ptr<Node<_T>>;
+    using size_type = size_t;
+
+public:
+    enum class PathType {Whole, Part};
```

Comment:
Whenever using braces, always have a space after the opening brace, and before the closing one.